beluga/include/beluga/algorithm/estimation.hpp

```diff
@@ -171,7 +172,34 @@ std::pair<Sophus::SE2<Scalar>, Sophus::Matrix3<Scalar>> estimate(Poses&& poses,
 
 /// Computes mean and covariance Returns a pair consisting of the estimated mean pose and its covariance.
 /**
- * Given a range of 2D poses, computes the estimated pose by averaging the translation
+ * Given a range of poses, computes the estimated pose and the standard deviation by averaging the translation.
+ *
+ * \tparam Poses A [sized range](https://en.cppreference.com/w/cpp/ranges/sized_range) type whose
+ *  value type is `std::vecotr<Scalar>`.
+ * \tparam Weights A [sized range](https://en.cppreference.com/w/cpp/ranges/sized_range) type whose
+ *  value type is `Scalar`.
+ * \tparam Pose The pose value type of the given range.
+ * \param poses Range of 1D poses.
+ * \param weights Range of weights.
+ * \return The estimated pose and its standard deviation.
+ */
+template <
+    class Poses,
+    class Weights,
+    class Pose = ranges::range_value_t<Poses>,
+    typename = std::enable_if_t<std::is_arithmetic_v<Pose>>>
+std::pair<Pose, Pose> estimate(Poses&& poses, [[maybe_unused]] Weights&& weights) {
+  // TODO(alon): use the weights to calculate the weighted average.
```

Comment:
@Alondruck meta: if we are taking weights, we should compute weighted mean and [variance](https://www.itl.nist.gov/div898/software/dataplot/refman2/ch2/weighvar.pdf).